browser/ntp_background/ntp_tab_helper.cc

```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ntp_background/ntp_tab_helper.h"
+
+#include "brave/browser/ntp_background/view_counter_service_factory.h"
+#include "brave/components/ntp_background_images/browser/view_counter_service.h"
+#include "chrome/browser/profiles/profile.h"
+#include "content/public/browser/navigation_handle.h"
+
+namespace ntp_background_images {
+
+NTPTabHelper::NTPTabHelper(content::WebContents* web_contents)
+    : WebContentsObserver(web_contents),
+      content::WebContentsUserData<NTPTabHelper>(*web_contents) {
+  Profile* profile =
+      Profile::FromBrowserContext(web_contents->GetBrowserContext());
+  if (profile) {
+    view_counter_service_ = ViewCounterServiceFactory::GetForProfile(profile);
+  }
+}
+
+NTPTabHelper::~NTPTabHelper() = default;
+
+void NTPTabHelper::DidStartNavigation(
+    content::NavigationHandle* navigation_handle) {
```

Comment:
url which was acquired in `DidStartNavigation` can be changed by redirects. Is it intentional? If not, we can get the url during `DidFinishNavigation` call